src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -156,20 +188,35 @@ private boolean isMuleEE() {
 
   // it checks both elements have the same attributes
   private boolean shouldUpdateApikitConfig(Element apikitConfigFromApi, Element apikitConfigFromMuleConfig) {
-    if (apikitConfigFromMuleConfig == null) {
-      return false;
-    }
-    for (Attribute attr : apikitConfigFromApi.getAttributes()) {
-      Attribute muleConfigAttr = apikitConfigFromMuleConfig.getAttribute(attr.getName());
-      if (muleConfigAttr == null) {
-        return true;
+    boolean shouldUpdate = apikitConfigFromMuleConfig == null;
+    if (!shouldUpdate) {
+      for (Attribute attr : apikitConfigFromApi.getAttributes()) {
+        Attribute muleConfigAttr = apikitConfigFromMuleConfig.getAttribute(attr.getName());
+        if (muleConfigAttr == null || attributeHasChanged(attr.getValue(), muleConfigAttr.getValue())) {
+          return true;
+        }
       }
+    }
+    return shouldUpdate;
+  }
 
-      if (!attr.getValue().equals(muleConfigAttr.getValue())) {
-        return true;
-      }
+  private boolean attributeHasChanged(String currentAttribute, String incomingAttribute) {
+    Optional<String> normalizedCurrentAttribute = normalizePath(currentAttribute);
+    Optional<String> normalizedIncomingAttribute = normalizePath(incomingAttribute);
+    boolean hasChanged = false;
+    boolean attributesExist = normalizedCurrentAttribute.isPresent() && normalizedIncomingAttribute.isPresent();
+    if (attributesExist) {
+      hasChanged = !normalizedCurrentAttribute.get().contains(normalizedIncomingAttribute.get());
```

Comment:
Optionals overcomplicated this method. Without them, this method just needs to return the comparison of normalized paths.

Reply:
why not return ?